### DISCUSSION TITLE
Mlflow.evaluate: custom metric function without eval_df

### DISCUSSION THREAD

**Discussion by khojarohan4** [mlflow/mlflow#6470] [Q&A]
I want to create a custom function that can be called when mlflow.evaluate is invoked. From my understanding, any custom function takes 3 arguments namely eval_df,builtin_metrics and artifact path of which artifact path is optional.

I looked through the comprehensive example but couldnt find an example to what I'm trying to do.

I want the below function to be called when passed to custom metrics which works fine if define as below. Here X_test passed inside the function is referenced globally instead of been passed as a part of the function. Is there a better way to handle this?

I dont need the target or prediction from the eval_df as it can be seen but I pass this as argument given what is expected.

Example function:

def brier_score(eval_df,builtin_metrics):
    return {"Brier_Loss": brier_score_loss(X_test['target'], model.predict_proba(X_test.drop('target', axis=1))[:,1])}

**Comment by khojarohan4** [comment]
@WeichenXu123 Can I take few minutes of your time to get your inputs on this one please?

**Reply by khojarohan4** [reply]
@WeichenXu123 Possible for you to take look at it once? Struggling to find a way to get this to work.

**Reply by WeichenXu123** [reply]
Sorry for late response,

I don't think you use the custom metric in correct way.

`custom metric` is designed to compute metric over the dataset passed to `mlflow.evaluate`, an example is:
https://github.com/mlflow/mlp-regression-template/blob/6ea7425ef3200e7fb09188b326362ba6d5f244ca/steps/custom_metrics.py

And when calling `mlflow.evaluate`, set the `custom_metrics` argument to include your custom metric function.
